pandas/core/groupby/groupby.py

```diff
@@ -2070,17 +2070,10 @@ def shift(self, periods=1, freq=None, axis=0):
     def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                    axis=0):
         """Calcuate pct_change of each value to previous entry in group"""
-        if freq is not None or axis != 0:
-            return self.apply(lambda x: x.pct_change(periods=periods,
-                                                     fill_method=fill_method,
-                                                     limit=limit, freq=freq,
-                                                     axis=axis))
-
-        filled = getattr(self, fill_method)(limit=limit).drop(
-            self.grouper.names, axis=1)
-        shifted = filled.shift(periods=periods, freq=freq)
-
-        return (filled / shifted) - 1
+        return self.apply(lambda x: x.pct_change(periods=periods,
```

Comment:
Isn't this essentially eliminating a vectorized solution across the board? How are the ASVs comparing?

Reply:
pct_change is blacklisted in benchmarks/groupby.py.